src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3705,26 +3704,30 @@ void ride_construction_toolupdate_entrance_exit(ScreenCoordsXY screenCoords)
     gMapSelectFlags |= MAP_SELECT_FLAG_ENABLE;
     gMapSelectFlags |= MAP_SELECT_FLAG_ENABLE_ARROW;
     gMapSelectType = MAP_SELECT_TYPE_FULL;
-    gMapSelectPositionA.x = x;
-    gMapSelectPositionA.y = y;
-    gMapSelectPositionB.x = x;
-    gMapSelectPositionB.y = y;
-    gMapSelectArrowDirection = direction_reverse(direction);
-    gMapSelectArrowPosition.x = x;
-    gMapSelectArrowPosition.y = y;
-    gMapSelectArrowPosition.z = _unkF44188.z * 8;
+    gMapSelectPositionA.x = entranceOrExitCoords.x;
+    gMapSelectPositionA.y = entranceOrExitCoords.y;
+    gMapSelectPositionB.x = entranceOrExitCoords.x;
+    gMapSelectPositionB.y = entranceOrExitCoords.y;
+    gMapSelectArrowDirection = direction_reverse(entranceOrExitCoords.direction);
+    gMapSelectArrowPosition.x = entranceOrExitCoords.x;
+    gMapSelectArrowPosition.y = entranceOrExitCoords.y;
+    gMapSelectArrowPosition.z = entranceOrExitCoords.z * 8;
     map_invalidate_selection_rect();
 
-    direction = direction_reverse(gRideEntranceExitPlaceDirection);
+    entranceOrExitCoords.direction = direction_reverse(gRideEntranceExitPlaceDirection);
     stationNum = gRideEntranceExitPlaceStationIndex;
-    if (!(_currentTrackSelectionFlags & TRACK_SELECTION_FLAG_ENTRANCE_OR_EXIT) || x != gRideEntranceExitGhostPosition.x
-        || y != gRideEntranceExitGhostPosition.y || direction != gRideEntranceExitGhostPosition.direction
+    if (!(_currentTrackSelectionFlags & TRACK_SELECTION_FLAG_ENTRANCE_OR_EXIT)
+        || entranceOrExitCoords.x != gRideEntranceExitGhostPosition.x
```

Comment:
Doesn't CoordsXYZD have a definition for the `==` method, so that you can do `entranceOrExitCoords == gRideEntranceExitGhostPosition`?

Reply:
It does not. The only struct on `Location.hpp` that defines the `operator==` is `TyleCoordsXYZ`

Reply:
Then perhaps it's a good idea to introduce it.

Reply:
Will do